homeassistant/components/media_player/samsungtv.py

```diff
@@ -247,6 +250,22 @@ def media_previous_track(self):
         """Send the previous track command."""
         self.send_key('KEY_REWIND')
 
+    def play_media(self, media_type, media_id, **kwargs):
+        """Support changing a channel."""
+        if media_type != MEDIA_TYPE_CHANNEL:
+            raise NotImplementedError()
+
+        # media_id should only be a channel number
+        try:
+            if int(media_id) < 1:
+                raise ValueError()
+        except ValueError:
+            raise TypeError('Media ID must be a positive integer')
+
+        for digit in media_id:
+            self.send_key('KEY_' + digit)
+            time.sleep(KEY_PRESS_TIMEOUT)
```

Comment:
Is this absolutely necessary ?

Reply:
we could make it configurable?

Reply:
But then again, commands also can't be sent too quickly to the Samsung TV, otherwise you can get a broken pipe. 🤷‍♂️

Reply:
I think I know why ...

Why does every key press start a new remote ?

https://github.com/mlebrun/home-assistant/blob/a33be8573db15ba9b4f859b130d9d42e027f5420/homeassistant/components/media_player/samsungtv.py#L161

I don't own a samsungtv but maybe rewrite this code not to use "get_remote()" for every single key press...

Reply:
It only configures the remote on the first attempt: https://github.com/mlebrun/home-assistant/blob/a33be8573db15ba9b4f859b130d9d42e027f5420/homeassistant/components/media_player/samsungtv.py#L146 and saves it off to a class member. So really it's just saying "if we don't already have a remote, get a new one". so in theory, it shouldn't only need to get it once and then it has the instance to use for subsequent requests. also, if the pipe breaks or another issue occurs, it'll destroy the remote instance and then it uses the same logic to get a fresh remote.

Reply:
You are right, I miss looked the get_remote() function.

Reply:
We don't want to use sync sleep. If sleep is necessary, convert the `play_media` method to async and use `asyncio. sleep`. That's free to use without penalty to our thread pool. Schedule the call to `send_key`on the executor via `hass.async_add_job` within the async method.

Reply:
Will do!